pkg/instrumentation/exporter.go

```diff
@@ -0,0 +1,150 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package instrumentation
+
+import (
+	"fmt"
+	"path/filepath"
+
+	corev1 "k8s.io/api/core/v1"
+
+	"github.com/open-telemetry/opentelemetry-operator/apis/v1alpha1"
+	"github.com/open-telemetry/opentelemetry-operator/internal/naming"
+	"github.com/open-telemetry/opentelemetry-operator/pkg/constants"
+)
+
+func configureExporter(exporter v1alpha1.Exporter, pod *corev1.Pod, container *corev1.Container) {
+	if exporter.Endpoint != "" {
+		if getIndexOfEnv(container.Env, constants.EnvOTELExporterOTLPEndpoint) == -1 {
+			container.Env = append(container.Env, corev1.EnvVar{
+				Name:  constants.EnvOTELExporterOTLPEndpoint,
+				Value: exporter.Endpoint,
+			})
+		}
+	}
+	if exporter.TLS == nil {
+		return
+	}
+	// the name cannot be longer than 63 characters
+	secretVolumeName := naming.Truncate("%s-secret-%s", 63, volumeName, exporter.TLS.SecretName)
+	secretMountPath := fmt.Sprintf("/otel-auto-instrumentation-secret-%s", exporter.TLS.SecretName)
+	configMapVolumeName := naming.Truncate("%s-configmap-%s", 63, volumeName, exporter.TLS.ConfigMapName)
+	configMapMountPath := fmt.Sprintf("/otel-auto-instrumentation-configmap-%s", exporter.TLS.ConfigMapName)
+
+	if exporter.TLS.CA != "" {
```

Comment:
should we require that if one of these is set, all of them should be set?

Reply:
There are three valid combinations:

1. ca
2. key, secret
3. ca, key, secret

The rule here is - if the key is defined the secret must be as well. However I would not implement validation on this to give users more flexibility. Eg. cert/key already defined in the workload or consumed from env var.

Reply:
>  However I would not implement validation on this to give users more flexibility. Eg. cert/key already defined in the workload or consumed from env var.

I'm fine with that but I would show a warning in the webhook validator.

Reply:
added